core/src/main/java/software/amazon/awssdk/core/exception/SdkServiceException.java

```diff
@@ -39,208 +36,177 @@
 @SdkPublicApi
 public class SdkServiceException extends SdkException {
 
-    /**
-     * The unique  identifier for the service request the caller made. The
-     * request ID can uniquely identify the request and can be used by the
-     * downstream service to identify specific caller requests to help
-     * with debugging.
-     */
-    private String requestId;
-
-    /**
-     * The error message as returned by the service.
-     */
-    private String errorMessage;
-
-    /**
-     * The error code represented by this exception.
-     */
-    @ReviewBeforeRelease("Is this too specific to AWS?")
-    private String errorCode;
-
-    /**
-     * {@link ErrorType} is a best effort determination of whether
-     * or not a given {@link SdkServiceException} was caused by a
-     * problem with the downstream service or a problem with the clients
-     * request.
-     * <p>
-     * Requests marked as {@link ErrorType#CLIENT} indicate a problem
-     * with the request that the caller made. These requests are not expected
-     * to be able to succeed on a retry.
-     * <p>
-     * Requests marked with {@link ErrorType#SERVICE} indicate a problem
-     * on the service side. These requests may be able to succeed on a retry.
-     * <p>
-     * Requests that can't be determined to be either service or client errors
-     * are marked as {@link ErrorType#UNKNOWN}. These requests may be able to be
-     * safely retried.
-     */
-    private ErrorType errorType = ErrorType.UNKNOWN;
+    private final String requestId;
+    private final int statusCode;
 
-    /**
-     * The HTTP status code that was returned with this error.
-     */
-    private int statusCode;
+    protected SdkServiceException(Builder b) {
+        super(b);
+        this.requestId = b.requestId();
+        this.statusCode = b.statusCode();
+    }
 
     /**
-     * The name of the service that sent this error response.
+     * The requestId that was returned by the called service.
+     * @return String containing the requestId
      */
-    private String serviceName;
+    public String requestId() {
+        return requestId;
+    }
 
     /**
-     * All of HTTP headers that were returned in the response.
+     * The status code that was returned by the called service.
+     * @return int containing the status code.
      */
-    private Map<String, String> headers;
+    public int statusCode() {
```

Comment:
Do we need this here if it's also available in the http response in error details?